frontend/javascripts/maintenance_banner.tsx

```diff
@@ -4,123 +4,141 @@ import {
-    );
-    if (currentMaintenance == null) return;
-    return (
-      <Alert
-        message={
-          <>
-            Currently under maintenance, scheduled until{" "}
-            <FormattedDate timestamp={currentMaintenance.endTime} />. {currentMaintenance.message}
-          </>
-        }
-        type="warning"
-        banner
-        onMouseEnter={() => {
-          if (isInAnnotationView) {
-            toggleTopOrBottomPosition();
-          }
-        }}
-        style={{ ...position, position: isInAnnotationView ? "absolute" : "sticky" }}
-      />
+  return (
+    <Alert
+      message={
+        <>
+          Currently under maintenance, scheduled until <FormattedDate timestamp={endTime} />.{" "}
+          {message}
+        </>
+      }
+      type="warning"
+      banner
+      style={BANNER_STYLE}
+    />
+  );
+}
+
+export function MaintenanceBanner() {
+  const activeUser = useSelector((state: OxalisState) => state.activeUser);
+
+  const [closestUpcomingMaintenance, setClosestUpcomingMaintenance] = useState<
+    MaintenanceInfo | undefined
+  >(undefined);
+  const [currentMaintenance, setCurrentMaintenance] = useState<MaintenanceInfo | undefined>(
+    undefined,
+  );
+
+  async function pollMaintenances() {
+    const newScheduledMaintenances = await listCurrentAndUpcomingMaintenances();
+
+    const closestUpcomingMaintenance = newScheduledMaintenances
+      .filter((maintenance) => maintenance.startTime > Date.now())
+      .filter(
+        (maintenance) =>
+          maintenance.id !== activeUser?.novelUserExperienceInfos.latestAcknowledgedMaintenanceInfo,
+      )
+      .sort((a, b) => a.startTime - b.startTime);
+
+    const currentMaintenance = newScheduledMaintenances.find(
+      (maintenance) => maintenance.startTime < Date.now(),
     );
-  };
 
-  if (currentAndUpcomingMaintenances.length === 0) return null;
-  const currentlyUnderMaintenanceBanner = getCurrentMaintenanceBanner();
-  if (currentlyUnderMaintenanceBanner != null) {
-    return currentlyUnderMaintenanceBanner;
+    setCurrentMaintenance(currentMaintenance);
+    setClosestUpcomingMaintenance(_.first(closestUpcomingMaintenance));
   }
-  const upcomingMaintenanceBanners = getClosestUpcomingMaintenanceBanner();
-  return upcomingMaintenanceBanners == null ? null : upcomingMaintenanceBanners;
+
+  useEffect(() => {
+    if (currentMaintenance || closestUpcomingMaintenance) {
+      setNavbarHeight(constants.DEFAULT_NAVBAR_HEIGHT + constants.MAINTENANCE_BANNER_HEIGHT);
+    }
+
+    if (currentMaintenance === undefined && closestUpcomingMaintenance === undefined) {
+      // Reset Navbar height if maintenance is over
+      setNavbarHeight(constants.DEFAULT_NAVBAR_HEIGHT);
+    }
+  }, [currentMaintenance, closestUpcomingMaintenance]);
+
+  // Do an initial fetch of the maintenance status so that users are notified
+  // quickly in case of ongoing maintenances.
+  setTimeout(pollMaintenances, INITIAL_DELAY);
```

Comment:
```suggestion
  useEffect(() => {
    // Do an initial fetch of the maintenance status so that users are notified
    // quickly in case of ongoing maintenances.
    setTimeout(pollMaintenances, INITIAL_DELAY);
  }, [])
```

otherwise, a new timeout would be kicked off on each render.